csharp/ql/lib/semmle/code/csharp/frameworks/EntityFramework.qll

```diff
@@ -351,67 +352,137 @@ module EntityFramework {
 
     /** Holds if component stack `head :: tail` is required for the output specification. */
     predicate requiresComponentStackOut(
-      Content head, Type headType, SummaryComponentStack tail, int dist
+      Content head, Type headType, SummaryComponentStack tail, int dist,
+      DbContextClassSetProperty dbSetProp
     ) {
-      exists(Property dbSetProp, PropertyContent c1 |
+      exists(PropertyContent c1 |
         dbSetProp = this.getADbSetProperty(headType) and
         this.stepRev(c1, _, head, headType, 0) and
         c1.getProperty() = dbSetProp and
-        tail = SummaryComponentStack::jump(dbSetProp.getGetter()) and
+        tail = SummaryComponentStack::return() and
         dist = 0
       )
       or
       exists(Content tailHead, SummaryComponentStack tailTail, Type tailType |
-        this.requiresComponentStackOut(tailHead, tailType, tailTail, dist - 1) and
+        this.requiresComponentStackOut(tailHead, tailType, tailTail, dist - 1, dbSetProp) and
         tail = SummaryComponentStack::push(SummaryComponent::content(tailHead), tailTail) and
         this.stepRev(tailHead, tailType, head, headType, dist)
       )
     }
+
+    pragma[nomagic]
+    string getSyntheticNames(
+      SummaryComponentStack input, SummaryComponentStack output, DbContextClassSetProperty dbSet
```

Comment:
I think it would be better to replace `SummaryComponentStack input` with `Property mapped`, and get rid of the call to `input`; this removes a bit of recursion, as we only need to call `input` in `DbContextSaveChanges::propagatesFlow`.

Reply:
Yes, I will change that as well. We probably need to run DCA for EFCore to be on the safe side.